src/test/kotlin/org/virtuslab/pulumikotlin/codegen/step2intermediate/PulumiNameTest.kt

```diff
@@ -555,4 +555,64 @@ internal class PulumiNameTest {
         assertEquals("Provider", className)
         assertEquals("com.pulumi.equinixmetal", packageName)
     }
+
+    @Test
+    fun `a resource function name is decapitalized correctly`() {
+        // given
+        val token = "provider:module:OrganizationPolicy"
+        val namingConfiguration = PulumiNamingConfiguration.create(providerName = "provider")
+
+        // when
+        val pulumiName = PulumiName.from(token, namingConfiguration)
+        val namingFlags = NamingFlags(Root, Resource, Input, Java)
+        val resourceFunctionName = pulumiName.toResourceFunctionName(namingFlags)
+
+        // then
+        assertEquals("organizationPolicy", resourceFunctionName)
+    }
+
+    @Test
+    fun `a resource function name starting with a three-letter acronym is decapitalized correctly`() {
+        // given
+        val token = "provider:module:IAMBinding"
+        val namingConfiguration = PulumiNamingConfiguration.create(providerName = "provider")
+
+        // when
+        val pulumiName = PulumiName.from(token, namingConfiguration)
+        val namingFlags = NamingFlags(Root, Resource, Input, Java)
+        val resourceFunctionName = pulumiName.toResourceFunctionName(namingFlags)
+
+        // then
+        assertEquals("iamBinding", resourceFunctionName)
+    }
+
+    @Test
+    fun `a resource function name that is a three-letter acronym is decapitalized correctly`() {
+        // given
+        val token = "provider:module:CRL"
+        val namingConfiguration = PulumiNamingConfiguration.create(providerName = "provider")
+
+        // when
+        val pulumiName = PulumiName.from(token, namingConfiguration)
+        val namingFlags = NamingFlags(Root, Resource, Input, Java)
+        val resourceFunctionName = pulumiName.toResourceFunctionName(namingFlags)
+
+        // then
+        assertEquals("crl", resourceFunctionName)
+    }
+
+    @Test
+    fun `a resource function name starting with a two-letter acronym and number is decapitalized correctly`() {
+        // given
+        val token = "provider:module:EC2Fleet"
+        val namingConfiguration = PulumiNamingConfiguration.create(providerName = "provider")
+
+        // when
+        val pulumiName = PulumiName.from(token, namingConfiguration)
+        val namingFlags = NamingFlags(Root, Resource, Input, Java)
+        val resourceFunctionName = pulumiName.toResourceFunctionName(namingFlags)
+
+        // then
+        assertEquals("ec2Fleet", resourceFunctionName)
```

Comment:
Adding one more test for `VMwareCluster` might be nice, so that we are aware of this edge case.